src/Conversion/ToyToMidPass.cpp

```diff
@@ -174,6 +228,151 @@ class ToyBinaryPattern : public ConversionPattern {
 private:
   enum toy::LoweringPatternMode mode;
 };
+
+template <typename ToyBinaryOp, typename LoweredBinaryOp>
+class SimpleMatmulPattern : public ConversionPattern {
+public:
+  SimpleMatmulPattern(MLIRContext *ctx, toy::LoweringPatternMode mode)
+      : ConversionPattern(ToyBinaryOp::getOperationName(), 1, ctx), mode(mode) {
+  }
+
+  LogicalResult
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto loc = op->getLoc();
+    // todo:vector pattern is simply copied from addtensor
+    if (mode == toy::LoweringPatternMode::Vector) {
```

Comment:
Consider to remove this option. Just add `lowerMatmulOpToLoops` or even move codes in `lowerMatmulOpToLoops` to matchAndRewrite.